src/Microsoft.IdentityModel.Tokens/EventBasedLRUCache.cs

```diff
@@ -134,9 +111,8 @@ internal EventBasedLRUCache(
             _removeExpiredValues = removeExpiredValues;
             _compactIntervalInSeconds = compactIntervalInSeconds;
             _timeForNextExpiredValuesRemoval = DateTime.UtcNow.AddSeconds(_removeExpiredValuesIntervalInSeconds);
-            _timeForNextCompaction = DateTime.UtcNow.AddSeconds(_compactIntervalInSeconds);
-            _eventQueueTaskStopTime = DateTime.UtcNow;
             _maintainLRU = maintainLRU;
+            _eventQTask = Task.Run(EventQueueTaskAction);
```

Comment:
There are a few comments that need updated, for example, line 160 talks about processing until the queue is empty.

Reply:
Tried to address all comments.